src/main/java/org/primeframework/mvc/scope/BaseManagedCookieScope.java

```diff
@@ -114,10 +114,10 @@ protected Cookie processCookie(Cookie cookie, String fieldName, Class<?> type, T
       ThrowingFunction<byte[], String> oldFunction = r -> objectMapper.readerFor(String.class).readValue(r);
       ThrowingFunction<byte[], String> newFunction = r -> new String(r, StandardCharsets.UTF_8);
       if (compress || encrypt) {
-        cookie.value = CookieTools.fromCookie(cookieValue, true, encryptor, oldFunction, newFunction);
+        cookie.value = CookieTools.fromCookie(cookieValue, encrypt, encrypt, encryptor, oldFunction, newFunction);
```

Comment:
```suggestion
        cookie.value = CookieTools.fromCookie(cookieValue, encrypt, true, encryptor, oldFunction, newFunction);
```

Reply:
The reason this bothers me is that if a cookie has `compress && !encrypt`, the currently released code will attempt to decrypt a cookie in the old format. However, cookies in the old format do not support compression. That means that such a cookie would always fail parsing. I'm not sure whether it was intended to fail.

If we ignore the code and think about that scenario, it makes sense to me to allow the cookie to be parsed. If it is now expected to be compressed but not encrypted, and we are presented with a cookie that was written before compression was enabled, we can still return the value because compression is not related to security.